plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/FBTypeEditPart.java

```diff
@@ -38,38 +39,34 @@
 import org.eclipse.jface.resource.JFaceResources;
 import org.eclipse.jface.util.IPropertyChangeListener;
 import org.eclipse.swt.events.ControlListener;
-import org.eclipse.swt.widgets.Display;
 
 public class FBTypeEditPart extends AbstractConnectableEditPart {
 
 	private ControlListener controlListener;
 	final List<AbstractContainerElement> containerChildren = new ArrayList<>(6);
 	private DiagramFontChangeListener fontChangeListener;
 
-	private final Adapter adapter = new EContentAdapter() {
+	private final Adapter versionInfoAdapter = new AdapterImpl() {
 		@Override
 		public void notifyChanged(final Notification notification) {
 			super.notifyChanged(notification);
-			if (Notification.REMOVING_ADAPTER != notification.getEventType()) {
-				final Object feature = notification.getFeature();
-				if ((LibraryElementPackage.eINSTANCE.getVersionInfo().equals(feature))
-						|| (LibraryElementPackage.eINSTANCE.getVersionInfo_Version().equals(feature))) {
-					getFigure().updateVersionInfoLabel();
-				}
-
-				Display.getDefault().syncExec(() -> {
-					if ((null != getParent()) && (null != getFigure()) && (getFigure().isShowing())) {
-						refresh();
-					}
-				});
+			final Object feature = notification.getFeature();
+			if (notification.getEventType() == Notification.ADD
```

Comment:
I believe this would also need to cover the cases where a `VersionInfo` was `SET` or `REMOVE`d, as well as `ADD_MANY` and `REMOVE_MANY`.

Instead of handling all this manually, how about simply overriding `EContentAdapter.addAdapter(Notifier)` to self-add only to `LibraryElement` and `VersionInfo`?

Reply:
The main reason is that with EContentAdapter I'm never sure what is simply. But you are right doing all the cases by hand is way to cumbersome. Changed accordingly. However LibraryElement is not needed as the adapter is added to it as start of the recursion.